samza-core/src/main/scala/org/apache/samza/diagnostics/DiagnosticsStreamMessage.java

```diff
@@ -236,23 +228,15 @@ public static DiagnosticsStreamMessage convertToDiagnosticsStreamMessage(Metrics
    * @param containerModelMap map of container models to serialize.
    * @return
    */
-  private static Map<String, String> serializeContainerModelMap(Map<String, ContainerModel> containerModelMap) {
-    Map<String, String> serializedContainerModelMap = new HashMap<>();
+  private static String serializeContainerModelMap(Map<String, ContainerModel> containerModelMap) {
```

Comment:
Do you have a javadoc somewhere explaining why you need do serialize explicitly instead of having Jackson do it on its own? (i.e. need to use default typing for backwards compatibility which doesn't work with some of the Samza data models like Partition since it doesn't have an empty constructor)
If you don't have that detail somewhere, could you please add it somewhere in here? This will be useful context for someone who is looking into this code in the future.

Reply:
added